src/v/cluster/controller_stm.cc

```diff
@@ -25,6 +25,9 @@ ss::future<> controller_stm::on_batch_applied() {
     if (!_feature_table.is_active(features::feature::controller_snapshots)) {
         co_return;
     }
+    if (!_snapshot_enabled()) {
+        co_return;
+    }
```

Comment:
just wondering if anything bad would happen if someone started toggling this flag